configure.ac

```diff
@@ -1405,15 +1423,30 @@ if test "$Py_LTO" = 'true' ; then
       case $ac_sys_system in
         Darwin*)
           # Any changes made here should be reflected in the GCC+Darwin case below
-          LTOFLAGS="-flto -Wl,-export_dynamic"
-          LTOCFLAGS="-flto"
+          if test $Py_LTO_POLICY = 'default'
+          then
+            LTOFLAGS="-flto -Wl,-export_dynamic"
+            LTOCFLAGS="-flto"
+          else
+            LTOFLAGS="-flto=${Py_LTO_POLICY} -Wl,-export_dynamic"
+            LTOCFLAGS="-flto=${Py_LTO_POLICY}"
+          fi
           ;;
         *)
-          LTOFLAGS="-flto"
+          if test $Py_LTO_POLICY = 'default'
+          then
+            LTOFLAGS="-flto"
+          else
+            LTOFLAGS="-flto={Py_LTO_POLICY}"
+          fi
           ;;
       esac
       ;;
     *gcc*)
+      if test $Py_LTO_POLICY = thin
```

Comment:
@holman
Should we check the full option also?
IMHO, `-flto` itself can be the alternative to the `full` option.

Reply:
I don't have strong opinions on this. Assuming `-flto` when `full` is passed as an option is a sensible assumption, since `full` is the default lto policy on clang, and no other compilers that I'm aware of have multiple lto implementations. 

It may lead to confusion in the future if, for example, clang changes default --flto policy to be thin, or if other compilers produce alternative lto implementations that have different naming schemes.  That said, unless anyone else has stronger opinions, I'd say leave it as is.

Reply:
agreed.